notebooks/roc_plot_generator.py

```diff
@@ -399,6 +399,15 @@ def get_col(df, col): # dumb util
 if TASK != "induction":
     d = {(d[0], d[1].hashable_tuple, d[2], d[3].hashable_tuple): False for d in exp.corr.all_edges()}
     d_trues = get_true_edges()
+    if ONLY_SAVE_CANONICAL and TASK == "ioi":
+        # Remove non-adjacent layer connections
+        def layer(name):
+            return int(name.split(".")[1])
+
+        for t in d_trues.keys():
+            if abs(layer(t[0]) - layer(t[2])) > 1:
```

Comment:
I suggest turning 

`            if abs(layer(t[0]) - layer(t[2])) > 1:`

into 

`if abs(layer(t[0]) - layer(t[2])) > 1 and ("hook_resid_mid" in [t[0], t[2]] or "mlp" in [t[0], t[2]]): # at least one of the nodes is an MLP node`

because currently I cannot tell whether the attention head connections are correct

Reply:
I solved this by plotting the heads and MLP+heads separately.